app/validation/validators.py

```diff
@@ -167,3 +167,31 @@ def __call__(self, form, from_field, to_field):
             raise validators.ValidationError(self.messages['INVALID_DATE_RANGE_TO_FROM_SAME'])
         elif from_date > to_date:
             raise validators.ValidationError(self.messages['INVALID_DATE_RANGE_TO_BEFORE_FROM'])
+
+
+class TimeInputCheck(object):
+    def __init__(self, messages=None):
+        if not messages:
+            messages = error_messages
+        self.messages = messages
+
+    def __call__(self, form, field):
+        logger.error(self.messages)
+        if form.minute.data and int(form.minute.data) > 59:
+            raise validators.ValidationError(self.messages['INVALID_MINUTES'])
+        if form.hour.data and int(form.hour.data) > 999:
+            raise validators.ValidationError(self.messages['INVALID_HOURS'])
+
+
+class TimeInputRequired(object):
```

Comment:
I wonder if this and the DateRequired should be consolidated into a single "FormResponseRequired" validator or similar, which checks that some data was entered at a form level (DateForm or TimeForm) in this case. Otherwise these type of validators will be added for each new form data structure that is added to the system.